### PR TITLE
Scrape Cryptocompare websocket data

### PR DESCRIPTION
**Pull request by IvanIvanoff**
## Changes

Subscribe to the CCCAGG Aggregate Index websocket endpoint of Cryptocompare and export the data to Kafka. In order to be able to use this subscription, the API key used must have an Enterprise subscription.

The data is exported into 2 different tables:
- The USD and BTC prices are exported to the currently used asset_prices_topic and they will end up in our existing database.
- The full data is exported to a new `asset_price_pairs` topic. It is called like this because now every coin will have a lot more trading pairs compared to just USD and BTC. Every record will contain data only for a given base/quote asset pair.
- The data in the new topic will be stored with its cryptocompare name instead of our slug. This is so we can export all the data we need without losing the data for which we don't have an asset. In this PR https://github.com/santiment/sanbase2/pull/2811 I've added the mapping of the known assets. We can add a CH dictionary that will be used to fetch the proper name in the queries.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [ ] I have added tests that prove my fix is effective or that my feature works

## Deployment steps
The Kafka topic has been created and the API keys are added here: https://github.com/santiment/devops/pull/2362

### CODE REVIEW

File: lib/sanbase/prices/point.ex [hunk 1]
```diff
@@ -0,0 +1,41 @@
+defmodule Sanbase.Price.Point do
```
Comment (tspenov) [inline lib/sanbase/prices/point.ex:1]: This looks like duplicate to `Sanbase.Cryptocompare.PricePoint`

File: mix.exs [hunk 2]
```diff
@@ -145,8 +148,8 @@ defmodule Sanbase.Mixfile do
       {:ueberauth_twitter, github: "santiment/ueberauth_twitter"},
       {:uuid, "~> 1.1"},
       {:vex, "~> 0.8.0", override: true},
-      {:ethereumex, "~> 0.7.0"},
-      {:ex_abi, "~> 0.5.4"}
+      {:waffle, "~> 1.1"},
```
Comment (tspenov) [inline mix.exs:151]: is waffle used ?

Comment (IvanIvanoff) [reply mix.exs:150]: Not sure. I just sort the deps in alphabetical order from time to time. Waffle was between 2 deps starting with `c`

File: lib/sanbase/prices/point.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/sanbase/prices/point.ex]: Yes, it is. First I though of putting all files in `sanbase/price` but then decided to go with a separate directory. This is a leftover.

File: lib/sanbase/cryptocompare/websocket_scraper.ex [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    export_data_point(point, last_points)
+
+    {:ok, %{state | last_points: last_points, msg_number: state.msg_number + 1}}
+  end
+
+  def handle_frame(%{"MESSAGE" => "FORCE_DISCONNECT"} = msg, _frame, state) do
+    Logger.warn("""
+    [CryptocompareWS] Received FORCE_DISCONNET for socket #{state.socket_id}. Reason: #{msg["INFO"]}
+    """)
+
```
Comment (tspenov) [inline lib/sanbase/cryptocompare/websocket_scraper.ex:120]: shouldn't `handle_disconnect` be called here ?

File: lib/sanbase/cryptocompare/websocket_scraper.ex [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+  defp get_slug_data_map() do
+    result =
+      Sanbase.Model.Project.SourceSlugMapping.get_source_slug_mappings("cryptocompare")
+      |> Map.new()
+
+    {:ok, result}
+  end
+
+  defp api_key(), do: Config.module_get(Sanbase.Cryptocompare, :api_key)
+  defp websocket_url(), do: "wss://streamer.cryptocompare.com/v2?api_key=#{api_key()}"
```
Comment (tspenov) [inline lib/sanbase/cryptocompare/websocket_scraper.ex:234]: we can maybe use the api key in header version of authentication ? secrets in query string looks more prone to be visible in logs and other places

File: lib/sanbase/cryptocompare/websocket_scraper.ex (hunk 3 above)
Comment (IvanIvanoff) [reply lib/sanbase/cryptocompare/websocket_scraper.ex:120]: `handle_disconnect/1` is a callback that is called from the library code. It is called in different scenarios: `:tcp_closed`, `:"$websockex_close_timeout"`, `:remote_close`, etc.

If we reach here we have received a :text message. This is properly pushed from the streamer (server) and it is followed by the server stopping the connection. This connection stop causes the `handle_disconnect` to be invoked.

I have tested this scenario the following way: Introduce a `Process.sleep(80000)` in the websocket library for the `:ping` frames and when the server does not receive 2 consecutive heartbeats this triggers the `handle_disconnect` callback.

File: lib/sanbase/cryptocompare/websocket_scraper.ex (hunk 4 above)
Comment (IvanIvanoff) [reply lib/sanbase/cryptocompare/websocket_scraper.ex]: Done

**Review by tspenov — APPROVED** [review]

**Pull request merged**